app/scripts/views/ready.js

```diff
@@ -152,16 +152,17 @@ define(function (require, exports, module) {
                  this.broker.hasCapability('syncPreferencesNotification'));
     },
 
-    afterRender: function () {
+    afterRender () {
       var graphic = this.$el.find('.graphic');
       graphic.addClass('pulse');
 
-      return this._createMarketingSnippet();
+      return this._createMarketingSnippet()
+        .then(proto.afterRender.bind(this));
     },
 
-    _createMarketingSnippet: function () {
+    _createMarketingSnippet () {
       if (! this.broker.hasCapability('emailVerificationMarketingSnippet')) {
-        return;
+        return p();
```

Comment:
I return the promise from this branch because the other branch returns a promise too.